src/modules/claim/claim.service.ts

```diff
@@ -379,4 +361,36 @@ export class ClaimService {
     const { sub: subject } = jwt.decode(token);
     return v5(JSON.stringify({ subject, role, version }), UUID_NAMESPACE);
   }
+
+  private async rolesByIssuer(issuer: string, namespace?: string): Promise<Role[]> {
+    const rolesOfIssuer = (await this.getBySubject({ subject: issuer, filters: { isAccepted: true } }))
+      .map((r) => r.claimType);
+    return (await this.roleService.getAll())
+      .filter((r: Role) => !namespace || r.namespace === namespace)
+      .filter(
+        (r) => r.definition.issuer.did?.includes(issuer) || rolesOfIssuer.includes(r.definition.issuer.roleName)
+      );
+  }
+
+  private async filterUserRelatedClaims(currentUser: string, qb: SelectQueryBuilder<Claim>) {
+    const ownedAssets = (await this.assetService.getByOwner(currentUser)).map((a) => a.id);
+    const offeredAssets = (await this.assetService.getByOfferedTo(currentUser)).map((a) => a.id);
+
+    const rolesByUser = (await this.rolesByIssuer(currentUser)).map((r) => r.namespace);
+
+    qb.andWhere(new Brackets(query => {
+      query.where(':currentUser = ANY (claim.claimIssuer)', {
+        currentUser,
+      })
+        .orWhere('claim.namespace IN (:...rolesByUser)', { rolesByUser })
```

Comment:
Looking again at `getBySubject()`, I think it will retrieve all of the approved claims that cache-server knows about for both off-chain and on-chain roles, right? So I don't think we even need to check the `claimIssuer` property anymore, right?
```suggestion
      query.where('claim.namespace IN (:...rolesByUser)', { rolesByUser })
```

Reply:
Hmm. I am not sure. Having `claim.namespace` in `rolesByIssuer` depends on role definitions and claims of current user, while presence of current user in `claim.claimIssuer` arbitrarily set by requester

Reply:
You are right, `claim.claimIssuer` filled from role issuers list. Since this property is of no use, maybe remove it?